configure-ironic.sh

```diff
@@ -76,6 +76,9 @@ set_http_basic_server_auth_strategy() {
 # Configure HTTP basic auth for ironic-api server
 if [ -f "${HTPASSWD_FILE}" ]; then
     set_http_basic_server_auth_strategy
+    # We've just set the DEFAULT auth_strategy to http_basic, we need to
+    # set the json_rpc auth_strategy back to noauth
+    crudini --set /etc/ironic/ironic.conf json_rpc auth_strategy noauth
```

Comment:
Sigh. Why don't we use authentication for JSON RPC?

Reply:
Why we need to set  to `noauth` if we are going to set to `http_basic`? .-. (At least this should only set things if `http_basic` is desired.

Reply:
Since https://github.com/metal3-io/ironic-image/pull/180 the json_rpc only happens via localhost, so I think we don't need auth, unlike the API where we do want the option of choosing http_basic?

Reply:
> Why we need to set to `noauth` if we are going to set to `http_basic`? .-. (At least this should only set things if `http_basic` is desired.

When we get to this point we have
DEFAULT.auth_strategy = http_basic

If we don't set 
json_rpc.auth_strategy = noauth 
it picks up the DEFAULT value




> Since #180 the json_rpc only happens via localhost, so I think we don't need auth, unlike the API where we do want the option of choosing http_basic?

yup, we're only listening on ::1
tcp6       0      0 ::1:8089                :::*                    LISTEN      97565/python3

Reply:
I think we should have authentication even if rpc is done only via localhost. Considering the pivoting use case, Ironic runs on a node where workload is also running and could also be in host networking, hence being able to access the json RPC interface. I would rather keep json_rpc authenticated.

Reply:
Yup, I'm happy to turn on auth for rpc, I'm not suggesting we don't do it, all I am trying to do here is fix the config so it works when auth is turned on for the ironic API.

Reply:
Authentication for json-rpc is [fully](https://github.com/metal3-io/ironic-image/blob/master/runironic-conductor.sh#L8-L10) [supported](https://github.com/metal3-io/ironic-image/blob/master/configure-ironic.sh#L93) by this image. It is a lot easier to set up if you have a version of ironic that includes [the fix to change the client configuration to the same format as the API clients](https://review.opendev.org/742474), but that is the only speedbump.